cmd/config/config.go

```diff
@@ -19,12 +21,20 @@ const (
 	UnsetFlag = "unset"
 )
 
-func NewConfigCmd() *cobra.Command {
+func NewConfigCmd(analyticsTracker analytics.Tracker) *cobra.Command {
 	cmd := &cobra.Command{
 		Long:  "View and modify specific configuration values",
 		RunE:  run(),
 		Short: "View and modify specific configuration values",
 		Use:   "config",
+		PreRun: func(cmd *cobra.Command, args []string) {
```

Comment:
Why is this PreRun while the others are PersistentPreRun?

Reply:
I considered leaving a comment. PersistentPreRun from my testing only ran on child cmds attached to the parent cmd. The parent cmd is where PersistentPreRun is defined. 

On the other hand PreRun will run on the parent command only. In this case config doesn't have any sub commands similar to the others (flags, environments). Hence why its a `PreRun` and not `PersistentPreRun`